http/bucket_service.go

```diff
@@ -309,24 +321,34 @@ func (h *BucketHandler) handlePostBucket(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	if err := h.BucketService.CreateBucket(ctx, req.Bucket); err != nil {
+	bucket, err := req.toInfluxDB()
+	if err != nil {
 		h.HandleHTTPError(ctx, err, w)
 		return
 	}
-	h.Logger.Debug("bucket created", zap.String("bucket", fmt.Sprint(req.Bucket)))
 
-	if err := encodeResponse(ctx, w, http.StatusCreated, newBucketResponse(req.Bucket, []*influxdb.Label{})); err != nil {
+	if err := h.BucketService.CreateBucket(ctx, bucket); err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+	h.Logger.Debug("bucket created", zap.String("bucket", fmt.Sprint(bucket)))
+
+	if err := encodeResponse(ctx, w, http.StatusCreated, newBucketResponse(bucket, []*influxdb.Label{})); err != nil {
 		logEncodingError(h.Logger, r, err)
 		return
 	}
 }
 
 type postBucketRequest struct {
-	Bucket *influxdb.Bucket
+	OrgID               influxdb.ID     `json:"orgID,omitempty"`
+	Name                string          `json:"name"`
+	Description         string          `json:"description"`
+	RetentionPolicyName string          `json:"rp,omitempty"` // This to support v1 sources
```

Comment:
What is to support `v1` sources? The entire field? Or the fact it's `omitempty`? Maybe clarify this comment if it's important to the reader.